File: src/main/java/cc/woverflow/hytils/handlers/lobby/armorstands/ArmorStandHider.java
/*
 * Hytils Reborn - Hypixel focused Quality of Life mod.
 * Copyright (C) 2020, 2021, 2022  Polyfrost, Sk1er LLC and contributors
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package cc.woverflow.hytils.handlers.lobby.armorstands;

import cc.polyfrost.oneconfig.utils.hypixel.HypixelUtils;
import cc.polyfrost.oneconfig.utils.hypixel.LocrawInfo;
import cc.polyfrost.oneconfig.utils.hypixel.LocrawUtil;
import cc.woverflow.hytils.HytilsReborn;
import cc.woverflow.hytils.config.HytilsConfig;
import cc.woverflow.hytils.handlers.cache.ArmorStandHandler;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.entity.item.EntityArmorStand;
import net.minecraftforge.client.event.RenderLivingEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

public class ArmorStandHider {
    @SubscribeEvent
    public void onEntityRenderer(RenderLivingEvent.Pre<EntityLivingBase> event) {
        final LocrawInfo locraw = LocrawUtil.INSTANCE.getLocrawInfo();
        if (HypixelUtils.INSTANCE.isHypixel() && locraw != null && (HytilsReborn.INSTANCE.getLobbyChecker().playerIsInLobby() && HytilsConfig.hideUselessArmorStands || HytilsConfig.hideUselessArmorStandsGame && (locraw.getGameType() == LocrawInfo.GameType.SKYBLOCK || locraw.getGameType() == LocrawInfo.GameType.BEDWARS || locraw.getGameType() == LocrawInfo.GameType.SKYWARS || locraw.getGameMode().contains("BRIDGE")))) {
            if (event.entity instanceof EntityArmorStand) {
                String unformattedArmorStandName = event.entity.getCustomNameTag().toLowerCase();
                for (String armorStands : ArmorStandHandler.INSTANCE.armorStandNames) {
                    if (unformattedArmorStandName.contains(armorStands)) {
                        event.setCanceled(true);
                        break;
                    }
                }
            }
        }
    }
}

File: src/main/kotlin/org/polyfrost/hytils/client/features/general/ArmorStandHider.kt
package org.polyfrost.hytils.client.features.general

import net.hypixel.data.type.GameType
import net.minecraft.ChatFormatting
import net.minecraft.client.renderer.entity.state.ArmorStandRenderState
import org.polyfrost.hytils.client.HytilsRebornConfig
import org.polyfrost.hytils.client.data.providers.ArmorStandData
import org.polyfrost.oneconfig.api.event.v1.events.RenderLivingEvent
import org.polyfrost.oneconfig.api.event.v1.invoke.impl.Subscribe
import org.polyfrost.oneconfig.api.hypixel.v1.HypixelUtils

object ArmorStandHider {
    @Subscribe
    fun onEntityRender(event: RenderLivingEvent.Pre) {
        if (!HytilsRebornConfig.isEnabled || event.entity !is ArmorStandRenderState || !HypixelUtils.isHypixel() || !shouldHide()) return

        val name = ChatFormatting.stripFormatting((event.entity as ArmorStandRenderState).nameTag?.string) ?: return
        if (ArmorStandData.armorStandNames.any { name.contains(it, ignoreCase = true) }) {
            event.cancelled = true
        }
    }

    private fun shouldHide(): Boolean {
        val location = HypixelUtils.getLocation()
        return when {
            HytilsRebornConfig.hideUselessArmorStands && !location.inGame() -> true
            HytilsRebornConfig.hideUselessArmorStandsGame && location.inGame() -> {
                location.gameType.orElse(null) in listOf(GameType.SKYBLOCK, GameType.BEDWARS, GameType.SKYWARS)
                        || location.mode.orElse("").contains("BRIDGE")
            }

            else -> false
        }
    }
}
